File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int n = s.length();
        List<String> ans = new ArrayList<>();

        for (int i = 0; i < n; i += k) {
            StringBuilder str = new StringBuilder();
            for (int j = i; j < i + k && j < n; j++) {
                str.append(s.charAt(j));
            }

            ans.add(str.toString());
        }

        int m = ans.get(ans.size() - 1).length();
        while (m++ < k) {
            ans.set(ans.size() - 1, ans.get(ans.size() - 1) + fill);
        }

        return ans.toArray(new String[0]);
    }
}

File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        int n = s.length();
        vector<string> ans;
        
        for(int i = 0; i < n; i += k) {
            string str = "";
            for(int j = i; j < i + k && j < n; j++) {
                str += s[j];
            }
            
            ans.push_back(str);
        }
        
        int m = ans.back().length();
        while(m++ < k) ans.back() += fill;
        
        return ans;
    }
};